testing/scenario_app/lib/src/platform_view.dart

```diff
@@ -26,21 +27,164 @@ List<int> _to64(num value) {
+
+  @override
+  void onBeginFrame(Duration duration) {
+    final SceneBuilder builder = SceneBuilder();
+
+    builder.pushOffset(0, 0);
+
+    // Create a path of rectangle with width of 200 and height of 300, starting from (100, 100).
+    //
+    // The left side of the rectangle becomes a symmetric curve towards the left.
+    // The right side of the rectangle becomes a double curve. From top of bottom of the curve, it goes left then right.
+    //     _______
+    //    |      |
+    //   |      |
+    //  |        |
+    //   |        |
+    //    |______|
+    Path path = Path();
+    path.moveTo(100, 100);
+    path.quadraticBezierTo(50, 250, 100, 400);
+    path.lineTo(350, 400);
+    path.cubicTo(400, 300, 300, 200, 350, 100);
+    path.close();
+    builder.pushClipPath(path);
+
+    finishBuilderByAddingPlatformViewAndPicture(builder, 3);
+  }
+}
+
+/// Platform view with transform.
+class PlatformViewTransformScenario extends PlatformViewScenario {
+  /// Constuct a platform view with transform scenario.
+  PlatformViewTransformScenario(Window window, String text, {int id = 0})
+      : super(window, text, id: id);
+
+  @override
+  void onBeginFrame(Duration duration) {
+    final SceneBuilder builder = SceneBuilder();
+
+    builder.pushOffset(0, 0);
+    final Matrix4 matrix4 = Matrix4.identity();
+    matrix4.rotateZ(1.0);
+    matrix4.scale(0.5, 0.5, 1.0);
+    matrix4.translate(1000.0, 100.0, 0.0);
+
+    final Float64List matrix4_64 =  Float64List.fromList(matrix4.storage);
+    builder.pushTransform(matrix4_64);
+
+    finishBuilderByAddingPlatformViewAndPicture(builder, 4);
+  }
+}
+
+/// Platform view with opacity.
+class PlatformViewOpacityScenario extends PlatformViewScenario {
+  /// Constuct a platform view with transform scenario.
+  PlatformViewOpacityScenario(Window window, String text, {int id = 0})
+      : super(window, text, id: id);
+
+  @override
+  void onBeginFrame(Duration duration) {
+    final SceneBuilder builder = SceneBuilder();
+
+    builder.pushOffset(0, 0);
+    builder.pushOpacity(150);
+
+    finishBuilderByAddingPlatformViewAndPicture(builder, 5);
+  }
+}
+
+mixin _BasePlatformViewScenarioMixin on Scenario {
+  int _textureId;
+
+  /// Construct the platform view related scenario
+  ///
+  /// It prepare a TextPlatformView so it can be added to the SceneBuilder in `onBeginFrame`.
+  /// Call this method in the constructor of the platform view related scenarios
+  /// to perform necessary set up.
+  void constructScenario(Window window, String text, int id) {
     const int _valueInt32 = 3;
     const int _valueFloat64 = 6;
     const int _valueString = 7;
     const int _valueUint8List = 8;
     const int _valueMap = 13;
     final Uint8List message = Uint8List.fromList(<int>[
       _valueString,
-      'create'.length, // this is safe as long as these are all single byte characters.
+      'create'
+          .length, // this is safe as long as these are all single byte characters.
```

Comment:
Please move this back up to line 186 - it's ok if the comment goes over 80. Maybe just ditch the comment, or rephrase as `// this won't work if we use multibyte characters`